### PR TITLE
fix: prevent updating the workspace repository configuration whenever no value is provided, so an existing configuration is not overwritten.

### LINKED ISSUE #1679: [AVM Module Issue]: Synapse Workspace deployment overwrites existing repository configuration
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/synapse/workspace

### (Optional) Module Version

0.1.0

### Description

the workspace repository configuration is overwritten whenever the workspacerepositoryconfiguration = {} and a workspace configuration is configured manually. also, when a configuration has been set with a previous deployment and the last published commit has not been provided in this configuration, the last published commit value will be empty resulting in an out-of-sync live synapse workspace in relation to the collaboration branch.

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by Kathedra**
prevent updating the workspace repository configuration whenever no value is provided, so an existing configuration is not overwritten.

## Pipeline Reference

| Pipeline |
| -------- |
|          |

## Type of Change

- [ ] Update to CI Environment or utlities (Non-module effecting changes)
- [x] Azure Verified Module updates:
  - [x] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] I have run `Set-AVMModule` locally to generate the supporting module files.
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

File: avm/res/synapse/workspace/main.bicep [hunk 1]
```diff
@@ -224,7 +224,7 @@ resource workspace 'Microsoft.Synapse/workspaces@2021-06-01' = {
       : null
     sqlAdministratorLogin: sqlAdministratorLogin
     sqlAdministratorLoginPassword: !empty(sqlAdministratorLoginPassword) ? sqlAdministratorLoginPassword : null
-    workspaceRepositoryConfiguration: workspaceRepositoryConfiguration
+    workspaceRepositoryConfiguration: !empty(workspaceRepositoryConfiguration) ? workspaceRepositoryConfiguration : null
```
Comment (AlexanderSehr) [inline avm/res/synapse/workspace/main.bicep:227]: One suggestion that should do the same, but is a bit simpler:
- Undo this change
- Make the parameter `workspaceRepositoryConfiguration` nullable (`param workspaceRepositoryConfiguration object?`

So if a user doesn't provide it, the value would be null and null would be passed into the deployment.

Comment (Kathedra) [reply avm/res/synapse/workspace/main.bicep:227]: when working locally, this may result in `Expected a value of type "object" but the provided value is of type "object | null`,  if, in the future, the workspacerepository configuration is split up with provided values and uses a user defined type. my fix prevents that. But now I tested this locally, it is save to say this is not an issue. will update the PR.

Comment (AlexanderSehr) [reply avm/res/synapse/workspace/main.bicep]: Hey Kathedra, I'm not sure I fully grasp the split you're referring to, but even then I would think it would be possible to handle it in a way that does not provoke a warning. In any case I appreciate that you already have UDTs on your mind. Looking forward to the time when we can import those bad boys...

File: avm/res/synapse/workspace/main.bicep [hunk 2]
```diff
@@ -74,7 +74,7 @@ param sqlAdministratorLogin string
 param sqlAdministratorLoginPassword string = ''
 
 @description('Optional. Git integration settings.')
-param workspaceRepositoryConfiguration object = {}
+param workspaceRepositoryConfiguration object ?
```
Comment (AlexanderSehr) [inline avm/res/synapse/workspace/main.bicep:77]:
```suggestion
param workspaceRepositoryConfiguration object?
```

The formatter will not like this space 😄

File: avm/res/synapse/workspace/main.bicep (hunk 1 above)
[the remark this answers is not in the document]
Comment (Kathedra) [reply avm/res/synapse/workspace/main.bicep]: Hi @AlexanderSehr , I ran Set-AVMModule. not sure if the versioning in main.json is correct, can you check that please? 
FYI I'm not very familiar with the WoW of Github (I primarily use DevOps). Let me know if it is correct now!
I do not have a pipeline badge. I can provide some proof if needed, but as you said, the change is small and will have no significant impact.

**Comment by Kathedra** [comment]
> @Kathedra please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Comment by Kathedra** [comment]
@microsoft-github-policy-service agree [company="{ilionx}"]

**Comment by Kathedra** [comment]
@microsoft-github-policy-service agree company="ilionx"

File: avm/res/synapse/workspace/main.bicep (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlexanderSehr) [reply avm/res/synapse/workspace/main.bicep]: Hey @Kathedra, looks about right. Regarding the badge I'll trust that it works (given the small scope). However, I would recommend you update your local bicep installation some time, as the changes in the `main.json` show that you're not using latest 😉

**Review by AlexanderSehr — APPROVED** [review]
